core/trino-main/src/main/java/io/trino/operator/MergeProcessorOperator.java

```diff
@@ -40,13 +41,20 @@ public static OperatorFactory createOperatorFactory(
             int rowIdChannel,
             int mergeRowChannel,
             List<Integer> redistributionColumns,
-            List<Integer> dataColumnChannels)
+            List<Integer> dataColumnChannels,
+            Function<Page, Page> pagePreprocessor)
     {
-        MergeRowChangeProcessor rowChangeProcessor = createRowChangeProcessor(merge, rowIdChannel, mergeRowChannel, redistributionColumns, dataColumnChannels);
+        MergeRowChangeProcessor rowChangeProcessor = createRowChangeProcessor(merge, rowIdChannel, mergeRowChannel, redistributionColumns, dataColumnChannels, pagePreprocessor);
```

Comment:
It seems cleaner to pass the preprocessor into `Factory` and do this in the `MergeProcessorOperator` constructor lambda before calling `MergeRowChangeProcessor.transformPage()`. That way, it's called in one place for all (both) processors, rather than needing to do it in each one.